src/wallet/wallet.cpp

```diff
@@ -758,7 +758,7 @@ bool CWallet::IsSpent(const COutPoint& outpoint) const
             // check for the inverse condition and only consider spending
             // transactions in known, potentially active states.
             const auto& wtx = mit->second;
-            if (wtx.isConfirmed() || wtx.InMempool() || (wtx.isInactive() && !wtx.isAbandoned()))
+            if (!wtx.isAbandoned() && !wtx.isMempoolConflicted() && !wtx.isBlockConflicted())
```

Comment:
In commit "wallet: track mempool conflicts" (9fa7f813281d8520a5eb5adbd29fe3a0d2ab5a4f)

Would suggest deleting the "An output is considered spent..." comment above now that the code is simpler and it no longer adds any new information. The part of the comment about checking for the "inverse condition" is also no longer true, because the code is now checking directly if wtx is conflicted or abandoned, instead of checking for the inverse condition (that it is confirmed, or in mempool, or inactive and not abandoned).